pkg/mediationcontainer/client_websocket_transport.go

```diff
@@ -97,17 +103,24 @@ func (clientTransport *ClientWebSocketTransport) CloseTransportPoint() {
 	glog.V(4).Infof("[CloseTransportPoint] closing transport endpoint and listener routine")
 	clientTransport.closeRequested = true
 	// close listener
-	clientTransport.StopListenForMessages()
-	// close websocket
-	wsConn := clientTransport.ws
-	if wsConn != nil {
-		wsConn.Close()
-		wsConn = nil
+	clientTransport.stopListenForMessages()
+	clientTransport.closeAndResetWebSocket()
+}
+
+// Close current WebSocket connection and set it to nil.
+func (clientTransport *ClientWebSocketTransport) closeAndResetWebSocket() {
+	if clientTransport.status == Closed {
+		return
+	}
+	// close WebSocket
+	if clientTransport.ws != nil {
+		clientTransport.ws.Close()
```

Comment:
yeah, I can set it to nil. But is it really necessary? Because there is no other place to have logic that checks or requires clientTransport.ws should be nil or not...